RMQClient/RMQConnection.m

```diff
@@ -60,22 +60,19 @@
 #import <RMQFrame.h>
 #import <RMQProcessInfoNameGenerator.h>
 
-@interface RMQConnection ()
-@property (strong, nonatomic, readwrite) id <RMQTransport> transport;
-@property (nonatomic, readwrite) RMQReader *reader;
-@property (nonatomic, readwrite) id <RMQChannelAllocator> channelAllocator;
-@property (nonatomic, readwrite) id <RMQFrameHandler> frameHandler;
+@property (weak, nonatomic, readwrite) id <RMQTransport> transport;
+@property (nonatomic, weak, readwrite) RMQReader *reader;
+@property (nonatomic, weak, readwrite) id <RMQChannelAllocator> channelAllocator;
+@property (nonatomic, weak, readwrite) id <RMQFrameHandler> frameHandler;
 @property (nonatomic, readwrite) id<RMQLocalSerialQueue> commandQueue;
 @property (nonatomic, readwrite) id<RMQWaiterFactory> waiterFactory;
-@property (nonatomic, readwrite) id<RMQHeartbeatSender> heartbeatSender;
-@property (nonatomic, weak, readwrite) id<RMQConnectionDelegate> delegate;
+@property (nonatomic, weak, readwrite) id<RMQHeartbeatSender> heartbeatSender;
 @property (nonatomic, readwrite) id <RMQChannel> channelZero;
-@property (nonatomic, readwrite) RMQConnectionConfig *config;
+@property (nonatomic, weak, readwrite) RMQConnectionConfig *config;
```

Comment:
Adding this was an error on my part.
If config is nil the handshake fails.

Removing weak from config still fixes the retain cycle.